### PR TITLE
add jaxrs hostname verification tests

### PR DESCRIPTION
**Pull request by WhiteCat22**
Adding some new tests for testing Hostname Verification with JAX-RS Clients

### CODE REVIEW

**Review by neuwerk — DISMISSED** [review]
LGTM!

**Comment by WhiteCat22** [comment]
#libby #build

**Comment by WhiteCat22** [comment]
#libby #build

**Comment by WhiteCat22** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0.client_fat

File: dev/com.ibm.ws.jaxrs.2.0.client_fat/test-applications/hostnameverification/src/io/openliberty/jaxrs/client/fat/hostnameverification/servlet/HostnameVerificationClientTestServlet.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+import componenttest.annotation.SkipForRepeat;
+import componenttest.app.FATServlet;
+
+/*
+ * Test the different combinations of SSL Hostname Verification config in the server.xml
+ * and the JAX-RS "com.ibm.ws.jaxrs.client.disableCNCheck" client property.
+ * 
+ * Skip the tests for `restfulWS-3.0`+ since there is no current way to disable 
+ * hostname verificaiton with RESTEasy.
```
Comment (jim-krueger) [inline dev/com.ibm.ws.jaxrs.2.0.client_fat/test-applications/hostnameverification/src/io/openliberty/jaxrs/client/fat/hostnameverification/servlet/HostnameVerificationClientTestServlet.java:37]: Have you talked with James Perkins about this?   Wouldn't it be in RESTEasy's best interest to support this?

**Review by jim-krueger — COMMENTED** [review]
Looks good.  Just a quick question

File: dev/com.ibm.ws.jaxrs.2.0.client_fat/test-applications/hostnameverification/src/io/openliberty/jaxrs/client/fat/hostnameverification/servlet/HostnameVerificationClientTestServlet.java (hunk 1 above)
Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.2.0.client_fat/test-applications/hostnameverification/src/io/openliberty/jaxrs/client/fat/hostnameverification/servlet/HostnameVerificationClientTestServlet.java:37]: I haven't, probably could. This is just to figure out what the heck works/doesn't currently.

Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.2.0.client_fat/test-applications/hostnameverification/src/io/openliberty/jaxrs/client/fat/hostnameverification/servlet/HostnameVerificationClientTestServlet.java:37]: Honestly, I thought this was merged back in May, didn't realize it was still hanging out.

**Comment by WhiteCat22** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0.client_fat

**Comment by WhiteCat22** [comment]
#libby #build #spawn.fullfat.buckets=com.ibm.ws.jaxrs.2.0.client_fat

**Review by jim-krueger — APPROVED** [review]

**Review by neuwerk — APPROVED** [review]
LGTM!

**Pull request merged**